js/src/threepio.js

```diff
@@ -1,48 +1,79 @@
 import * as tf from '@tensorflow/tfjs';
 import mappedCommands from '../../static/mapped_commands.json';
+import { translationMissing } from './_errors';
 import { NORMALIZATION_REGEX } from './_constants';
+import Command from './command';
 
 export default class Threepio {
-  constructor(from, version) {
-    this.form = from;
+  constructor(from, to, version) {
+    this.fromLang = from;
+    this.toLang = to;
     this.version = version;
     this.mappedCommands = mappedCommands;
     this.tf = tf;
   }
 
-  normalizeFunctionName(name) {
-    return name
+  _normalizeFunctionName(name, lang) {
+    const result = name
       .match(NORMALIZATION_REGEX)
       .join('')
       .toLowerCase();
+
+    if (result in this.mappedCommands[lang]) {
+      return result;
+    }
+
+    throw new Error(translationMissing(name));
   }
 
-  matchArgs(args, argMap) {
-    // Sort args by index
-    argMap.args.sort((a, b) =>
-      a.index === b.index ? 0 : a.index < b.index ? -1 : 1
-    );
-    for (const [i, arg] of args.entries()) {
-      // TODO: match args based on fromLibrary name
-      // TODO: Ensure order is preserved
-      // TODO: Ensure there are not any missing required arguments
-      console.log(i, arg);
-      continue;
+  orderArgs(cmd, fromInfo, toInfo) {
```

Comment:
This is the logic for parsing the location of the all of the args and kwargs, and re-ordering them to match the formatting of the command we are translating to.